eng/deployment/product-construction-service-deploy.ps1

```diff
@@ -42,39 +77,52 @@ else
+        if ($sleep -eq $true) 
+        {
+            Start-Sleep -Seconds 60
+        }
+        $newRevisionRunningState = az containerapp revision show --name $containerappName --resource-group $resourceGroupName --revision $newRevisionName --query "properties.runningState"
+        Write-Host "New revision running state: $newRevisionRunningState"
+        $sleep = $true
+    } While ($newRevisionRunningState -notmatch "Running" -and $newRevisionRunningState -notmatch "Failed")
+
+    if ($newRevisionRunningState -match "Running") {
+        Write-Host "Assigning label $inactiveLabel to the new revision"
+        # assign the label to the new revision
+        az containerapp revision label add --label $inactiveLabel --name $containerappName --resource-group $resourceGroupName --revision $newRevisionName | Out-Null
+
+        # transfer all traffic to the new revision
+        az containerapp ingress traffic set --name $containerappName --resource-group $resourceGroupName --label-weight "$inactiveLabel=100" | Out-Null
+        Write-Host "All traffic has been redirected to label $inactiveLabel"
+    }
+    else {
+        Write-Host "New revision is not running. Check revision $newRevisionName logs in the inactive revisions. Deactivating the new revision"
+        az containerapp revision deactivate --revision $newRevisionName --name $containerappName --resource-group $resourceGroupName
+        exit 1
     }
-    $newRevisionRunningState = az containerapp revision show --name $containerappName --resource-group $resourceGroupName --revision $newRevisionName --query "properties.runningState"
-    Write-Host "New revision running state: $newRevisionRunningState"
-    $sleep = $true
-} While ($newRevisionRunningState -notmatch "Running" -and $newRevisionRunningState -notmatch "Failed")
-
-if ($newRevisionRunningState -match "Running") {
-    Write-Host "Assigning label $inactiveLabel to the new revision"
-    # assign the label to the new revision
-    az containerapp revision label add --label $inactiveLabel --name $containerappName --resource-group $resourceGroupName --revision $newRevisionName | Out-Null
-
-    # transfer all traffic to the new revision
-    az containerapp ingress traffic set --name $containerappName --resource-group $resourceGroupName --label-weight "$inactiveLabel=100" | Out-Null
-    Write-Host "All traffic has been redirected to label $inactiveLabel"
 }
-else {
-    Write-Host "New revision is not running. Check revision $newRevisionName logs in the inactive revisions. Deactivating the new revision"
-    az containerapp revision deactivate --revision $newRevisionName --name $containerappName --resource-group $resourceGroupName
-    exit 1
-}
\ No newline at end of file
+finally {
+    # Start the service. This either starts the new revision or the old one if the new one failed to start
+    Write-Host "Starting the product construction service"
+    $pcsStartUrl = $pcsUrl + "/status/start"
```

Comment:
We could use the status url here.

Or maybe better. Just define 3 variables on top of the file - status, start and stop URLs and use them through out the script?

Reply:
Done